sdk/core/System.ClientModel/src/ModelReaderWriter/CollectionBuilder.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Provides an interface to define how to construct, add items, and produce the final collection type.
+/// </summary>
+public abstract class CollectionBuilder
+{
+    /// <summary>
+    /// Gets the builder instance of the collection type.
+    /// </summary>
+    protected internal abstract object GetBuilder();
```

Comment:
It's a little strange that the method is called GetBuilder and the type is also called CollectionBuilder. It would be nice to differentiate between the two somehow.

Reply:
Resolved with the refactor on CollectionBuilder to make it internal